docs/adr/0021-move-verification-tests.md

```diff
@@ -0,0 +1,25 @@
+# 1. Avoid listing file counts unless user is authenticated
+
+Date: 2024-08-12
+
+## Status
+
+Accepted
+
+## Context
+
+Regular CI runs (on push, PR etc.) ran all tests, including the verification tests.
+
+These verification tests made requests to the GitHub API.
+
+Lots of CI runs in close succession could result in this API returning 403 errors (see #4434), causing test failures.
```

Comment:
I would be a bit more assertive here and say that it `does result` in the API returning 403 errors rather than `could`. 

It's also caused some test failures when two workflows are run concurrently (i.e two of us push to different PRs as the same time as happened with Steve and I recently) because we share state in the opensafely-testing GitHub org.